querycsv/querycsv.py

```diff
@@ -89,29 +87,29 @@ def read_sqlfile(filename):
     # Currently (11/11/2007) this routine knows only two things about SQL:
     #    1. Lines that start with "--" are comments.
     #    2. Lines that end with ";" terminate a SQL statement.
-    sqlfile = open(filename, "rt")
-    sqlcmds = []
-    currcmd = ''
-    for line in sqlfile:
-        line = line.strip()
-        if len(line) > 0 and not (len(line) > 1 and line[:2] == "--"):
-            currcmd = "%s %s" % (currcmd, line)
-            if line[-1] == ';':
-                sqlcmds.append(currcmd.strip())
-                currcmd = ''
-    return sqlcmds
+    with open(filename, "rt") as sqlfile:
+        sqlcmds = []
+        currcmd = ''
+        for line in sqlfile:
+            line = line.strip()
+            if len(line) > 0 and not (len(line) > 1 and line[:2] == "--"):
```

Comment:
it would be better if you change `len(line) > 1 and line[:2] == "--"` to `line.startswith('--')`